### ISSUE TITLE
Build fails with compilation error: webkit-2.4.11

### ISSUE THREAD

**Issue by xelxebar** [NixOS/nixpkgs#30782]
## Issue description

When attempting to install `nixpkgs.uzbl` (currently `uzbl-v0.9.0`), the dependency `webkit-2.4.11` starts a build but fails with a compilation error. The relevant errors from the build log are below:

```
Source/JavaScriptCore/API/JSStringRef.cpp: In function 'OpaqueJSString* JSStringCreateWithCharacters(const JSChar*, size_t)':
Source/JavaScriptCore/API/JSStringRef.cpp:40:50: error: invalid conversion from 'const JSChar* {aka const short unsigned int*}' to 'const UChar* {aka const char16_t*}' [-fpermissive]
     return OpaqueJSString::create(chars, numChars).leakRef();
                                                  ^
In file included from Source/JavaScriptCore/API/JSStringRef.cpp:31:0:
Source/JavaScriptCore/API/OpaqueJSString.h:49:39: note:   initializing argument 1 of 'static WTF::PassRefPtr<OpaqueJSString> OpaqueJSString::create(const UChar*, unsigned int)'
     static PassRefPtr<OpaqueJSString> create(const UChar* characters, unsigned length)
                                       ^~~~~~
Source/JavaScriptCore/API/JSStringRef.cpp: In function 'OpaqueJSString* JSStringCreateWithCharactersNoCopy(const JSChar*, size_t)':
Source/JavaScriptCore/API/JSStringRef.cpp:65:83: error: invalid conversion from 'const JSChar* {aka const short unsigned int*}' to 'const UChar* {aka const char16_t*}' [-fpermissive]
     return OpaqueJSString::create(StringImpl::createWithoutCopying(chars, numChars)).leakRef();
                                                                                   ^
In file included from ./Source/WTF/wtf/text/WTFString.h:29:0,
                 from Source/JavaScriptCore/API/OpaqueJSString.h:31,
                 from Source/JavaScriptCore/API/JSStringRef.cpp:31:
./Source/WTF/wtf/text/StringImpl.h:393:54: note:   initializing argument 1 of 'static WTF::PassRef<WTF::StringImpl> WTF::StringImpl::createWithoutCopying(const UChar*, unsigned int)'
     WTF_EXPORT_STRING_API static PassRef<StringImpl> createWithoutCopying(const UChar* characters, unsigned length);
                                                      ^~~~~~~~~~~~~~~~~~~~
Source/JavaScriptCore/API/JSStringRef.cpp: In function 'const JSChar* JSStringGetCharactersPtr(JSStringRef)':
Source/JavaScriptCore/API/JSStringRef.cpp:86:30: error: invalid conversion from 'const UChar* {aka const char16_t*}' to 'const JSChar* {aka const short unsigned int*}' [-fpermissive]
     return string->characters();
            ~~~~~~~~~~~~~~~~~~^~
make[1]: *** [GNUmakefile:70162: Source/JavaScriptCore/API/libjavascriptcoregtk_1_0_la-JSStringRef.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/tmp/nix-build-webkitgtk-2.4.11.drv-0/webkitgtk-2.4.11'
make: *** [GNUmakefile:25838: all] Error 2
builder for ‘/nix/store/2pdl996ffn441x75b90prjaq6pv5lf2j-webkitgtk-2.4.11.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/jjb34awh2fjygj711p18pm7lcq40kshw-uzbl-v0.9.0.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/jjb34awh2fjygj711p18pm7lcq40kshw-uzbl-v0.9.0.drv’ failed
```

### Steps to reproduce

The build requires an entry in `permittedInsecurePackages`:

    $ cat ~/.config/nixpkgs/config.nix
    {
      permittedInsecurePackages = [
        "webkitgtk-2.4.11"
      ];
    }

and on my machine fails a few hours after initiating the install of `uzbl`:

    $ nix-env -i -A nixpkgs.uzbl

I doubt uzbl is relevant but include those details here for completeness.

## Technical details

    $ lsb_release -a
    LSB Version:	1.0
    Distributor ID:	VoidLinux
    Description:	Void Linux
    Release:	rolling
    Codename:	void

    $ nix-env --version
    nix-env (Nix) 1.11.4

    $ nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion
    "18.03pre118061.69607d7662"

    $ grep build-use-sandbox /etc/nix/nix.conf
    <returns 1>

## Notes

Apologies if this issue belongs somewhere else or is some trivial mistake on my part. I'm still not very familiar with the nix system. I'm willing to provide any further details if necessary.

**Comment by peterhoeg** [comment]
Fixed in #30623 which hasn't hit ```nixpkgs-unstable``` yet.

You *really* shouldn't be using a browser that depends on ```webkitgtk24```.

**Comment by xelxebar** [comment]
Oh shoot. I didn't realize that issue was relevant. Thanks for the swift handling.
